ThingIFSDK/ThingIFSDK/ScheduleOncePredicate.swift

```diff
@@ -3,11 +3,11 @@
 //  ThingIFSDK
 //
 //  Created by syahRiza on 5/10/16.
-//  Copyright © 2016 Kii. All rights reserved.
+//  Copyright (c) 2016 Kii. All rights reserved.
 //
 
 /** Class represents ScheduleOncePredicate. */
-open class ScheduleOncePredicate: Predicate {
+open class ScheduleOncePredicate: NSObject, Predicate {
```

Comment:
@yongpingchen Opposite to input data, we decided to implement `NSCoding` to output data. `NSCoding` is strongly connected to `Foundation` framework. If we do not extends `NSObject` to a class implementing `NSCoding`, it becomes very hard to use.

I wrote following blogs why hard to use if a class implementing `NSCoding` does not extends `NSObject`:
* https://kiicorp.docbase.io/posts/185776?list=%2Fmy%2Fposts&q=author%3Ashinsukenara
* https://kiicorp.docbase.io/posts/185834?list=%2Fmy%2Fposts&q=author%3Ashinsukenara